tracer/src/Datadog.Trace/Ci/Agent/CIAgentlessWriter.cs

```diff
@@ -145,26 +162,48 @@ private static async Task InternalFlushEventsAsync(object state)
                             watermarkCompletion = watermarkEvent.CompletionSource;
                             break;
                         }
-                        else if (ciTestCycleBuffer.CanProcessEvent(item))
+
+                        // ***
+                        // Force a flush by time (When the queue was never empty in a complete BatchInterval period)
+                        // ***
+                        await Task.WhenAll(
+                            buffers.FlushCiTestCycleBufferWhenTimeElapsedAsync(BatchInterval),
+                            buffers.FlushCiCodeCoverageBufferWhenTimeElapsedAsync(BatchInterval)).ConfigureAwait(false);
+
+                        // ***
+                        // Add the item to the right buffer, and flush the buffer in case is needed.
+                        // ***
+                        if (ciTestCycleBuffer.CanProcessEvent(item))
                         {
                             // The CITestCycle endpoint can process this event, we try to add it to the buffer.
+                            // If the item cannot be added to the buffer but the buffer has events
+                            // we assume that is full and needs to be flushed.
                             while (!ciTestCycleBuffer.TryProcessEvent(item) && ciTestCycleBuffer.HasEvents)
                             {
-                                // If the item cannot be added to the buffer but the buffer has events
-                                // we assume that is full and needs to be flushed.
-                                await writer.SendPayloadAsync(ciTestCycleBuffer).ConfigureAwait(false);
-                                ciTestCycleBuffer.Clear();
+                                await buffers.FlushCiTestCycleBufferAsync().ConfigureAwait(false);
                             }
+
+                            continue;
+                        }
+
+                        if (ciCodeCoverageBuffer.CanProcessEvent(item))
+                        {
```

Comment:
Should this be an `else if`?

Reply:
Can be, but the `continue` in the previous block makes the job. I wanted this way because I can separate the logic for each buffer in two separate blocks. just for the visuals/styling